docs/apps/r-env.md

```diff
@@ -0,0 +1,133 @@
+
+## Licenses
+
+Information on licenses that are in use for R and associated software (such as packages) can be found on the [R Project website](https://www.r-project.org/Licenses/).
+
+## Usage
+
+#### Loading the module
+
+To use the default version of this module on Puhti, initialize it with:
+
+```
+module load r-env
+```
+
+Note that Puhti login nodes are not intended for heavy computing. To use R in Puhti, please either request an interactive job in a computing node or submit a non-interactive batch job in Slurm. 
+
+#### Interactive use
+
+To interactively use R on Puhti's computational nodes, run this command after initializing the `r-env` module. This will request a single-processor R session. Note that you will need to modify the requested duration, memory, project ID and partition as required:
+
+```
+srun -n1 -t hh:mm:ss --x11=first --mem=4G --pty --account=project_id -p partition R --no-save 
+```
+
+For information on available partitions, see [here](https://github.com/CSCfi/csc-user-guide/blob/master/docs/computing/running/batch-job-partitions.md).
+
+**Note:** RStudio is currently unavailable on Puhti, but is included in future installation plans.
+
+#### Non-interactive use
+
+You can also run R scripts non-interactively using batch job files. This is particularly useful for jobs that require multiple cores or a lot of memory. See this [link](https://github.com/CSCfi/csc-user-guide/blob/master/docs/computing/running/creating-job-scripts.md) for detailed information on how to prepare batch jobs. Information on submitting array jobs can be found [here](https://github.com/CSCfi/csc-user-guide/blob/master/docs/computing/running/array-jobs.md).
+
+Below is an example for submitting a single-processor R batch job on Puhti. Note that the `test` partition is used here (which has a time limit of 15 minutes and is used for testing purposes only).
+
+```
+#!/bin/bash -l
+#SBATCH --job-name=r_single_proc
+#SBATCH --account=<project>
+#SBATCH --output=output_%j.txt
+#SBATCH --error=errors_%j.txt
+#SBATCH --partition=test
+#SBATCH --time=00:05:00
+#SBATCH --ntasks=1
+#SBATCH --nodes=1
+#SBATCH --mem-per-cpu=1000
+
+module load r-env/3.6.1
+srun Rscript --no-save myrscript.R
+```
+
+#### R package installations
+
+It is possible to check if a particular package is already installed as follows.
+
+```r
+# One way is to try loading the package:
+library(packagename)
+
+# If you don't want to load the package, it is also
+# possible to search through a list:
+installed_packages <- library()$results[,1]
+"packagename" %in% installed_packages
+
+# Note: both ways are sensitive to upper- and lower-case letters
+```
+
+Additional R package installations can be arranged via two separate routes:
+
+- Project-specific installations can be used by creating a separate package directory in the `/projappl/<project>` directory (instructions below; also see [here](https://docs.csc.fi/#computing/disk/#projappl-directory) for information on ProjAppl)
```

Comment:
`docs.csc.fi` should be changed to internal markdown links. Example `https://docs.csc.fi/#computing/disk/#projappl-directory` becomes `../computing/disk.md#projappl-directory`. This way they can be automatically checked for validity.